pkg/ingester/client/client.go

```diff
@@ -25,12 +26,19 @@ type HealthAndIngesterClient interface {
 type closableHealthAndIngesterClient struct {
 	IngesterClient
 	grpc_health_v1.HealthClient
-	conn *grpc.ClientConn
+	conn    *grpc.ClientConn
+	metrics *Metrics
+	addr    string
 }
 
 // MakeIngesterClient makes a new IngesterClient
 func MakeIngesterClient(addr string, cfg Config, metrics *Metrics) (HealthAndIngesterClient, error) {
-	dialOpts, err := cfg.GRPCClientConfig.DialOption(grpcclient.Instrument(metrics.RequestDuration))
+	unary, stream := grpcclient.Instrument(metrics.requestDuration)
```

Comment:
Note from testing: this wraps the `HealthClient` methods with a circuit breaker too which is probably not what we want.

```
ts=2023-08-04T19:51:10.560584491Z caller=pool.go:196 level=warn msg="removing ingester failing healthcheck" addr=XXX:9095 reason="circuit breaker is open"
```

Reply:
The circuit breaker interceptor knows the `method` too. What if configure the circuit breaker to only act on a list of configured methods, so that we're explicit on which gRPC methods we want the circuit breaker to apply?

Reply:
What do you think about only applying the interceptor to methods that contain `cortex.Ingester`? This would avoid the need to keep a list of methods up to date and allow non-ingester things like the health check to bypass it. Or would you prefer and explicit list of methods?

Reply:
I've added of methods to circuit-break in e312654368dd17058cf575f524d612cecade594c, LMK what you think.